third_party/move/move-compiler/src/cfgir/cfg.rs

```diff
@@ -192,6 +192,72 @@ impl<'a> BlockCFG<'a> {
             println!();
         }
     }
+
+    /// Adds a block on any critical edge: an edge from a node with multiple successors to
+    /// a node with multiple predecessors.  Returns true if any changes occur; in this
+    /// case, caller must call recompute().
+    pub fn remove_critical_edges<Ftype>(&mut self, mut make_label: Ftype) -> bool
+    where
+        Ftype: FnMut() -> Label,
+    {
+        let critical_edges: Vec<_> = self
+            .successor_map
+            .iter()
+            .filter(|(_, nexts)| nexts.len() > 1)
+            .flat_map(|(lbl, nexts)| {
+                nexts
+                    .iter()
+                    .filter(|lbl2| {
+                        self.predecessor_map
+                            .get(lbl2)
+                            .map_or_else(|| false, |preds| (preds.len() > 1))
+                    })
+                    .map(|lbl2| (*lbl, *lbl2))
+            })
+            .collect();
+        for (pred, succ) in &critical_edges {
+            let new_label = make_label();
+            self.add_edge(*pred, *succ, new_label);
+        }
+        !critical_edges.is_empty()
```

Comment:
If you don't use vector, can use a bool in the loop to see whether its empty.

Reply:
Added a bool, but see above.